backend/src/services/socket/socket.service.ts

```diff
@@ -0,0 +1,64 @@
+import { Server as HttpServer } from 'http';
+import { Server as SocketServer } from 'socket.io';
+
+import { SocketEvent } from '~/common/enums/enums';
+import {
+  ChatMessageGetAllItemResponseDto,
+  Socket as SocketType,
+  SocketServerType,
+} from '~/common/types/types';
+
+class Socket {
```

Comment:
This service should be more abstract, we should be able to handle more than chat logic here. For chat we should use namespace

Reply:
I refactored some methods, but I decided to leave `sendMessage` method there, in order to have more explicit typing for the event. Should I leave it as it is or is it better to create methods, like `emit`, `listen` and accept event type and arguments/callbacks?

Reply:
It's better to create `emit` and `listen`